src/silo/common/date.cpp

```diff
@@ -0,0 +1,21 @@
+#include "silo/common/date.h"
+
+silo::common::Date silo::common::stringToDate(const std::string& value) {
+   auto split_position = value.find('-', 0);
+   auto split_position2 = value.find('-', split_position + 1);
+   const std::string year_string = value.substr(0, split_position);
+   const std::string month_string = value.substr(split_position + 1, split_position2);
+   const std::string day_string = value.substr(split_position2 + 1);
+   const uint32_t year = stoi(year_string);
+   const uint32_t month = stoi(month_string);
+   const uint32_t day = stoi(day_string);
+   const uint32_t date_value = (year << 16) + (month << 12) + day;
+   return Date{date_value};
+}
+
+std::string silo::common::dateToString(silo::common::Date date) {
+   const uint32_t year = date >> 16;
+   const uint32_t month = (date >> 12) & 0xF;
+   const uint32_t day = date & 0xFFF;
+   return std::to_string(year) + "-" + std::to_string(month) + "-" + std::to_string(day);
+}
```

Comment:
Please add a unit test (especially for error handling of `stringToDate` since this has to process user input.)